packages/angular-rspack/src/lib/plugins/index-html-plugin.ts

```diff
@@ -0,0 +1,161 @@
+import { addError, addWarning } from '../utils/rspack-diagnostics';
+import { urlJoin } from '../utils/url-join';
+import { addEventDispatchContract } from '../utils/index-file/add-event-dispatch-contract';
+
+export interface IndexHtmlPluginOptions extends IndexHtmlGeneratorOptions {
+  baseHref: string | undefined;
+  i18n: I18nOptions;
+  index: IndexExpandedDefinition;
+  isSsr: boolean;
+  outputPath: string;
+}
+
+const PLUGIN_NAME = 'IndexHtmlPlugin';
+
+export class IndexHtmlPlugin
+  extends IndexHtmlGenerator
+  implements RspackPluginInstance
+{
+  private _compilation: Compilation | undefined;
+  get compilation(): Compilation {
+    if (this._compilation) {
+      return this._compilation;
+    }
+
+    throw new Error('compilation is undefined.');
+  }
+
+  constructor(override readonly options: IndexHtmlPluginOptions) {
+    super(options);
+  }
+
+  apply(compiler: Compiler) {
+    compiler.hooks.thisCompilation.tap(PLUGIN_NAME, (compilation) => {
+      this._compilation = compilation;
+
+      compilation.hooks.processAssets.tapPromise(
+        {
+          name: PLUGIN_NAME,
+          stage: Compilation.PROCESS_ASSETS_STAGE_SUMMARIZE,
+        },
+        async () => {
+          const files: FileInfo[] = [];
+
+          try {
+            for (const chunk of compilation.chunks) {
+              for (const file of chunk.files) {
+                // https://github.com/web-infra-dev/rspack/blob/a2e1e21c7e1ed0f34e476ec270e3c5460c4a1a36/packages/rspack/src/config/defaults.ts#L606
+                if (
+                  file.endsWith('.hot-update.js') ||
+                  file.endsWith('.hot-update.mjs')
+                ) {
+                  continue;
+                }
+
+                files.push({
+                  name: chunk.name,
+                  file,
+                  extension: extname(file),
+                });
+              }
+            }
+
+            const outputPaths = ensureOutputPaths(
+              this.options.outputPath,
+              this.options.i18n
+            );
+
+            for (const [locale, outputPath] of outputPaths.entries()) {
+              const { csrContent, warnings, errors } = await this.process({
+                files,
+                outputPath,
+                baseHref:
+                  this.getLocaleBaseHref(locale) ?? this.options.baseHref,
+                lang: locale || undefined,
+              });
+
+              let html = csrContent;
+              if (this.options.isSsr) {
+                html = await addEventDispatchContract(csrContent);
+              }
+
+              const { RawSource } = compiler.rspack.sources;
+              compilation.emitAsset(
+                getIndexOutputFile(this.options.index),
+                new RawSource(html)
+              );
+              warnings.forEach((msg) => addWarning(compilation, msg));
+              errors.forEach((msg) => addError(compilation, msg));
+            }
```

Comment:
If this is handling baseHref and lang per locale - do we still need this in the inline-i18n-plugin: https://github.com/nrwl/angular-rspack/pull/56/files#diff-09e27febaafdeed82c061631a6c612786a4990c9dac705c27080b09f1b78509cR91-R106